### PR TITLE
use Win API for date formatting

### LINKED ISSUE #1627: Yellow keycard is displayed in place of the blue skull key on Boom HUD
Tested in the [latest master artifact (x64)](https://github.com/fabiangreffrath/woof/actions/runs/8447502663)
The issue was introduced in this commit: https://github.com/fabiangreffrath/woof/commit/b32c4df60f0405184d357e1ec16488664206fa0a
Commenting out the line that specifies the locale fixes the issue:
https://github.com/fabiangreffrath/woof/blob/bb47d3f103b83555cb453f9b0ebd25267bffbf47/src/i_main.c#L55-L57

### PR DESCRIPTION
**Pull request by rfomin**
Fix #1627

### CODE REVIEW

File: src/mn_snapshot.c [hunk 1]
```diff
@@ -97,15 +105,31 @@ void MN_ReadSavegameTime(int i, char *name)
     }
     else
     {
+#if defined(_WIN32)
+        SYSTEMTIME lt;
+        wchar_t wdate[64];
+
+        GetLocalTime(&lt);
```
Comment (fabiangreffrath) [inline src/mn_snapshot.c:112]: Does this print the savegame file date?

Comment (rfomin) [reply src/mn_snapshot.c]: I am not sure what I was thinking, fixed here: https://github.com/fabiangreffrath/woof/pull/1631/commits/5f358670b454fa4e7675e95ae3d3a3d298ed5e32

File: src/mn_snapshot.c [hunk 2]
```diff
@@ -97,15 +106,49 @@ void MN_ReadSavegameTime(int i, char *name)
...
+        }
+        if (!GetFileTime(hFile, NULL, NULL, &ftWrite))
+        {
+            I_Printf(VB_ERROR, "MN_ReadSavegameTime: GetFileTime failed");
+            return;
+        }
+
+        FileTimeToSystemTime(&ftWrite, &stUTC);
+        SystemTimeToTzSpecificLocalTime(NULL, &stUTC, &stLocal);
+        GetDateFormatEx(LOCALE_NAME_USER_DEFAULT, DATE_SHORTDATE, &stLocal,
```
[the remark this answers is not in the document]
Comment (rfomin) [reply src/mn_snapshot.c]: Done here: https://github.com/fabiangreffrath/woof/pull/1631/commits/20a0e79784f55e1c5913cc622645f67edfce78b1

[the remark this answers is not in the document]
Comment (rfomin) [reply src/mn_snapshot.c]: > we could just s well stay with the "C" locale

Maybe we should, we don't have a Unicode font anyway.

[the remark this answers is not in the document]
Comment (fabiangreffrath) [reply src/mn_snapshot.c]: I have to admit, I'm not satisfied with any solution so far (not even my own). If the `setlocale()` call leads to issues on Windows, let's just guard it with `#ifndef _WIN32` and stay with the "C" locale for everything there.

[the remark this answers is not in the document]
Comment (rfomin) [reply src/mn_snapshot.c]: > If the setlocale() call leads to issues on Windows, let's just guard it with #ifndef _WIN32 and stay with the "C" locale for everything there.

OK, so should I close this PR? We can just commit straight to master with this change.

[the remark this answers is not in the document]
Comment (fabiangreffrath) [reply src/mn_snapshot.c]: Yes, please. Sorry for all the effort you put into this! But this is high above the amount of quirk I'd like to add for a minor issue like this. 


Does it help to add `setlocale(LC_ALL, "C")` right before `setlocale(LC_TIME, "")`? I can imagine this triggers the localized version of `toupper()` with the "C" locale and then sets the time locale properly?

[the remark this answers is not in the document]
Comment (rfomin) [reply src/mn_snapshot.c]: > Sorry for all the effort you put into this! 

There really isn't much effort involved. It's just that the Win API is very verbose. Most of the code is taken from MS examples.

> Does it help to add `setlocale(LC_ALL, "C")` right before `setlocale(LC_TIME, "")`? I can imagine this triggers the localized version of `toupper()` with the "C" locale and then sets the time locale properly?

It still won't work, here the analysis: https://github.com/fabiangreffrath/woof/pull/1630#issuecomment-2026917711 In short, any call to `setlocale` changes the behaviour of `toupper` to a non-ANSII version. Working alternative is `setlocale(LC_ALL, ".UTF-8");`

**Comment by rfomin** [comment]
Fixed here: https://github.com/fabiangreffrath/woof/commit/4aab690db399f9022f7972c42cd4b8e7681094d7

**Pull request closed without merging**